File: javaFile/delete_at_position.java
class Node {
    int data;
    Node next;
    Node(int data) {
        this.data = data;
        this.next = null;
    }
}

class LinkedList {
    Node head;
    LinkedList() {
        this.head = null;
    }

    void createList(int data){
        Node newNode = new Node( data);
        if (head == null){
            head = newNode;
            return;
        }
        Node current = head;
        while  (current.next != null) {
            current = current.next;
        }
        current.next = newNode;

    }

    void deleteAtPosition(int position) {
        if (head == null) {
            System.out.println("List is empty");
            return;
        }
        if (position == 1) {
            head = head.next;
            return;
        }
        Node current = head;
        for (int i = 1; i < position - 1; i++) {
            if (current == null || current.next == null) {
                System.out.println("Position out of bounds");
                return;
            }
            current = current.next;
        }
        if (current.next == null) {
            System.out.println("Position out of bounds");
            return;
        }
        current.next = current.next.next;
    }
    void printList() {
        Node temp = head;
        while ( temp != null) {
            System.out.println(temp.data+ "-> ");
            temp = temp.next;
        }
    }

public class delete_at_position {

    public static void main(String[] args) {
        LinkedList list = new LinkedList();
        list.createList(1);
        list.createList(2);
        list.createList(3);
        list.createList(4);
        list.createList(5);
        System.out.println("Original List:");
        list.printList();
        
        int positionToDelete = 3;
        list.deleteAtPosition(positionToDelete);
        
        System.out.println("List after deleting at position " + positionToDelete + ":");
        list.printList();
    }
}

File: python/deleteAtPosition.py
#traverse from 1 to position - 1
#delete node = temp.next
#temp.next = delete.next
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

class LinkedList:
    def __init__(self):
        self.head = None

    def create_list(self, data):
        new_node = Node(data)
        if self.head is None:
            self.head = new_node
            return
        current = self.head
        while current.next:
            current = current.next
        
        current.next = new_node
    
    def deleteAtPosition(self, position):
        if self.head is None:
            print("list is empty")
            return
        if position == 1:
            if self.head.next is None:
                self.head = None
                return
            temp = self.head
            self.head = self.head.next
            del temp
            return
        current = self.head
        for _ in range(1, position - 1):
            if current is None:
                print("Position out of bounds")
                return
            current = current.next

        if current is None or current.next is None:
            print("Position out of bounds")
            return
        delete_node = current.next
        current.next = delete_node.next
        del delete_node
        return
    
    def print_list(self):
        temp = self.head
        while temp:
            print(temp.data, end=" -> ")
            temp = temp.next

def main():
    linked_list = LinkedList()
    linked_list.create_list(1)
    linked_list.create_list(2)
    linked_list.create_list(3)
    linked_list.create_list(4)
    linked_list.create_list(5)
    print("Original List:")
    linked_list.print_list()
    position = 3
    linked_list.deleteAtPosition(position)
    print(f"List after deleting node at position {position}:")
    linked_list.print_list()
    print()

if __name__ == "__main__":
    main()

        
        
